.pre-commit-config.yaml

```diff
@@ -32,7 +50,10 @@ repos:
    - id: mixed-line-ending
      exclude: "\\.(sha|sha512|svg|vtk|vtp)$|\\/ThirdParty\\/|\\/Data\\/"
    - id: name-tests-test
+     exclude: "\\/ThirdParty\\/"
      args: ['--pytest-test-first']
+   - id: no-commit-to-branch
+     args: ['--branch','dashboard','--branch','python-builds','--branch','release','--branch','hooks','--branch', 'main','--branch','master','--pattern','release-*']
```

Comment:
Does this mean I will not be able to commit to a local branch named `master`? My typical workflow is to work in `master` branch, and push to topically-named branches on the internet, e.g.:
<img width="860" alt="Screenshot 2024-12-12 12 02 50" src="https://github.com/user-attachments/assets/9c5937a2-a7ab-4d0e-b2f1-63512f5fc647" />

Reply:
This is not a good practice that can result in a diverging `master`.

Reply:
You are correct.  Developers should not work on master branch for topics.  The motivation for preventing this can be seen in issue #4982.  Because it is working on the developers master branch, and they are trying to merge in their forks/master into ITK/master.    This makes it impossible for me to assist them with recommended code changes without breaking my un-corrupted master branch.

I am hoping that it will discourage PR's of topics on branches called "master"

I sometimes make mistake and commit temporary changes to my master branch, then those temporary changes corrupt other branches because my local "master" does not properly reflect the upstream/master or origin/master.

This is intended to keep those diverging masters from happening by forcing topic branches to have a meaningful name.

Reply:
Can we also, or instead of this, prevent opening PRs from people's `master` branch?

Reply:
I don't know how to do that.  Do you have a suggestion?

Reply:
I don't know whether it is even possible, much less how to do it 😄

Reply:
I don't think it would be possible.  I've never seen PR hooks like that.